src/zcash/IncrementalMerkleTree.hpp

```diff
@@ -21,8 +21,38 @@ class MerklePath {
 
     template <typename Stream, typename Operation>
     inline void SerializationOp(Stream& s, Operation ser_action, int nType, int nVersion) {
-        READWRITE(authentication_path);
-        READWRITE(index);
+        std::vector<std::vector<unsigned char>> pathBytes;
```

Comment:
Can I get some info on what this function is doing and why it's necessary specifically for the Darwin build?

Reply:
The problem is that a `bool` is ambiguous in terms of serialization, at least within the Bitcoin Core serialization framework. On GCC builds, it is serialized as a `uint8_t` (which is very inefficient), but Clang just refuses to compile it.

I have an improved version of this in #3107; I shall rebase this PR to replace the commit implementing this with it.

Reply:
What's the status of this: can this PR be merged as-is without #3107?

Reply:
I pulled in the commit from #3107, so this PR now exactly matches it.